### PR TITLE
[TTAHUB-2020] Recipient leadership on the RTR Profile tab

### PR DESCRIPTION
**Pull request by thewatermethod**
## Description of change

Add the "recipient leadership" table to the Profile tab of the Recipient record page

The table should only show:
- CFO
- Director of Head Start
- Director of Early Head Start

I had to update our existing script to parse the HSES data to access the CFO. We won't see CFO when this launches, but we should see it once this is deployed to production.

[The mockup](https://www.figma.com/proto/AKodYuKbKzxPmqtoIYkSuj/Profile-Spotlight?page-id=539%3A57933&type=design&node-id=539-61920&viewport=639%2C-64%2C0.13&t=kSeK4MJQLbMadUrJ-1&scaling=min-zoom&starting-point-node-id=539%3A59921) is here, but the Jira specified a few changes from the mockup.

- The pagination is removed, since we don't have historical data
- The widget is flipped to be on the top of the column
- The link to "grantee staff history" is removed, since we don't have historical data

## How to test

With a recent prod snapshot or data that includes imported ProgramPersonnel, view some recipients on the RTR and observe that they have director data. 

You'll see that multigrant recipients have some deduping going on. ([There is a good example here](/recipient-tta-records/1411/region/2/profile)) There shouldn't be multiple people with the same name AND the same title. It's ok, however, if there is Jim, Director for Early Head Start and a Jim, Director for Head Start. Or either of those and a Jim, CFO.

Make sure that it matches the mockup given the caveats above and that the data reflects what is present in the database and looks correct.

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-2020

### Every PR

- [ ] Meets issue criteria
- [ ] JIRA ticket status updated
- [ ] Code is meaningfully tested
- [ ] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [ ] API Documentation updated
- [ ] Boundary diagram updated
- [ ] Logical Data Model updated
- [ ] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions
- [ ] UI review complete

### Production Deploy

- [ ] Staging smoke test completed

### After merge/deploy

- [ ] Update JIRA ticket status

### CODE REVIEW

File: src/lib/updateGrantsRecipients.js [hunk 1]
```diff
@@ -71,8 +71,10 @@ async function getProgramPersonnel(grantId, programId, program) {
         lastName,
         suffix: getPersonnelField(currentRole, 'suffix', program),
         title: getPersonnelField(currentRole, 'title', program),
-        email,
-
+        email: getPersonnelField(currentRole, 'email', program),
```
Comment (AdamAdHocTeam) [inline src/lib/updateGrantsRecipients.js:74]: Good catch.

File: src/routes/recipient/handlers.js [hunk 2]
```diff
@@ -107,26 +128,36 @@ export async function searchRecipients(req, res) {
...
   try {
-    const { recipientId, regionId } = req.params;
-    // Check if user has access to this region.
-    const userId = await currentUserId(req, res);
-    const readRegions = await getUserReadRegions(userId);
-    if (!readRegions.includes(parseInt(regionId, 10))) {
-      res.sendStatus(403);
-      return;
-    }
+    const proceedQuestionMark = await checkAccessAndExistence(req, res);
```
Comment (AdamAdHocTeam) [inline src/routes/recipient/handlers.js:131]: proceed ?

**Review by AdamAdHocTeam — COMMENTED** [review]
When making the window smaller, might want to add some space to the right of the sort.

image

**Review by AdamAdHocTeam — COMMENTED** [review]
I could be wrong here but doesn't arrow pointing up mean AS (A-Z) seems backwards.

image

image

**Comment by thewatermethod** [comment]
> I could be wrong here but doesn't arrow pointing up mean ASC (A-Z) seems backwards.

Ascending does mean *up* and that is the way it is on the AR table.

**Comment by thewatermethod** [comment]
> When making the window smaller, might want to add some space to the right of the sort.

@AdamAdHocTeam ok, did so

**Review by AdamAdHocTeam — APPROVED** [review]
Looks good.

**Pull request merged**